core/shared/src/main/scala/it/unibo/scafi/core/RichLanguage.scala

```diff
@@ -92,6 +92,26 @@ trait RichLanguage extends Language { self: Core =>
             if (of1.compare(a._1, b._1) == 0) of2.compare(a._2, b._2) else of1.compare(a._1, b._1)
         }
 
+      implicit def tupleBounded4[T1, T2, T3, T4](implicit of1: Bounded[T1], of2: Bounded[T2], of3: Bounded[T3], of4: Bounded[T4]): Bounded[(T1, T2, T3, T4)] =
+        new Bounded[(T1, T2, T3, T4)] {
+          def top: (T1, T2, T3, T4) = (of1.top, of2.top, of3.top, of4.top)
+          def bottom: (T1, T2, T3, T4) = (of1.bottom, of2.bottom, of3.top, of4.top)
+          override def compare(a: (T1, T2, T3, T4), b: (T1, T2, T3, T4)): Int =
+            if (of1.compare(a._1, b._1) == 0) {
```

Comment:
Try to refactor this method by working on a `List(of1.compare(a._1,b._1), of2.compare(a._2,b._2), ...)` instead of nested `if/else`s.

Also, did you consider the possibility of not adding this method and instead adding dependency `stdlib-ext` to the test project or where features requiring `Bounded[A,B,C,D]` are to be used?